### PR TITLE
Fix many-to-many insert

### LINKED ISSUE #6: Failure when adding many-to-many association
## Description

We're encountering the following error when trying to add a new many-to-many association between two entities:

`TypeError: Cannot read property 'constructor' of undefined`

To be clear, we are *not* tracking changes on these entities (no `@TrackChanges` decorator), so getting an error from `nestjs-changelog` is quite unexpected. After some investigation, I figured out that the problem is a missing check for whether an entity is included in the event that notifies of a new insertion, here:

https://github.com/Shift3/nestjs-changelog/blob/b129bd2f788971f81e0bd7ffc383ccfc278fb321/lib/change.subscriber.ts#L26

So this problem will be seen anytime there's something being inserted into the database which has not been marked with the `@Entity` decorator, as is the case for the join table created by TypeORM for a many-to-many relationship.

## Workaround
One possibility, at least for this case, is to explicitly define an entity class for the join table (i.e., don't let TypeORM create it), which means you'll add the `@Entity` decorator, getting rid of the problem. However, this is [not a simple change].

The easiest way I found around this is to use `ChangeSubscriber.disableTracking` to wrap the `save()` call. But of course, this may not be desirable if you are saving things that you want tracking on.

[not a simple change]: https://orkhan.gitbook.io/typeorm/docs/many-to-many-relations#many-to-many-relations-with-custom-properties

### PR DESCRIPTION
**Pull request by pcnova**
## Changes

- Modified `ChangeSubscriber` to check whether an entity is being inserted.
- Added unit test (and related items) for the fix.

## Purpose

Closes #6, allowing consumers to save many-to-many relations.

### CODE REVIEW

File: lib/change.subscriber.ts [hunk 1]
```diff
@@ -23,7 +23,8 @@ export class ChangeSubscriber implements EntitySubscriberInterface<any> {
     if (ChangeSubscriber.disabled)
       return;
 
-    if (Reflect.hasMetadata("__track_changes", event.entity.constructor)) {
+    if (event.entity &&
+				Reflect.hasMetadata("__track_changes", event.entity.constructor)) {
```
Comment (mwallert) [inline lib/change.subscriber.ts:27]: Is this a linting error or purposely on a new line?

**Review by mwallert — APPROVED** [review]
I don't see any issues here, but I think @DropsOfSerenity has the most context for this repo.

File: lib/change.subscriber.ts (hunk 1 above)
Comment (pcnova) [reply lib/change.subscriber.ts:27]: It's a new line on purpose - since the second line is kind of long, I think it helps readability.

However, note that the weird, long spacing on that second line is an artifact of how GH renders tabs (this project uses tabs for indentation, not spaces). So, if you open this in VSCode, it actually looks like this:

Tabs look OK on VSCode

Comment (DropsOfSerenity) [reply lib/change.subscriber.ts:27]: Lets keep style based changes out of this PR for now, the next line is even longer, so this change would only make sense if we were to apply this style to the entire project.

The tab's don't really work well, it's usually tab for indentation and spaces for alignment.

File: lib/spec/change.spec.ts [hunk 2]
```diff
@@ -356,6 +358,20 @@ describe('Change Tracking', () => {
 
 				expect(changeCount).to.eq(1);
 			})
+
+			it("ignores many-to-many insert for untracked entity", async () => {
+				const entity = await unTrackedFactory.create();
+				const related = await relatedFactory.createMany(3);
+
+				// This inserts records into the join table implcitly created for MTM
+				// relationships. These should not be tracked, and should not fail either.
```
Comment (DropsOfSerenity) [inline lib/spec/change.spec.ts:367]: Should not fail should be expressed in test rather than comment. Probably we should use `chai-as-promised` here and do something like:

```typescript
await expect(entity.save()).to.be.fulfilled;
```

**Review by DropsOfSerenity — CHANGES_REQUESTED** [review]
Nice work catching that edge case! I just have a few changes to request before we can get this in. Also really appreciate the added tests!
